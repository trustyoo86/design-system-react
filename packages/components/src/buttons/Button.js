import React from 'react';
import { button } from '@dsr/styles';

const Button = (props) => {
  const { title, children } = props;

  return (
    <button styles={button}>
      { title ? title : children }
    </button>
  );
};

export default Button;
