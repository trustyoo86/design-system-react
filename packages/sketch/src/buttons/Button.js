'use strict';

import React from 'react';
import styled, { css } from '@emotion/primitives';
import { Text } from 'react-sketchapp';

function getColorSet(type = 'primary') {
  const basic = `
    background-color: #e55;
    color: #ffffff;
    border: 1px solid #e55;
  `;
  
  if (type === 'primary') { return basic; }
  if (type === 'bordered') {
    return `
      background-color: #ffffff;
      color: #c3c3c3;
      border: 1px solid #c3c3c3;
    `;
  }

  return basic;
}

const basic = css`
  box-sizing: border-box;
  padding: 0 5px;
  border-radius: 3px;
  margin-left: 5px;
  margin-right: 5px;
`;

const Buttons = {
  LargeButton: styled.View`
    ${basic}
    ${props => getColorSet(props.type)}
    width: 250px;
    height: 60px;
  `,
  LargeText: styled.Text`
    color: #ffffff;
    font-size: 18px;
  `,
  MediumButton: styled.View`
    ${basic}
    ${props => getColorSet(props.type)}
    width: 200px;
    height: 50px;
  `,
  MediumText: styled.Text`
    color: #ffffff;
    font-size: 16px;
  `,
  SmallButton: styled.View`
    ${basic}
    ${props => getColorSet(props.type)}
    width: 150px;
    height: 40px;
  `,
  SmallText: styled.Text`
    color: #ffffff;
    font-size: 14px;
  `,
  XSmallButton: styled.View`
    ${basic}
    ${props => getColorSet(props.type)}
    width: 100px;
    height: 30px;
  `,
  XSmallText: styled.Text`
    color: #ffffff;
    font-size: 12px;
  `,
};

function getComponents(props) {
  const { width } = props;
  const { 
    LargeButton,
    LargeText,
    MediumButton,
    MediumText,
    SmallButton,
    SmallText,
    XSmallButton,
    XSmallText,
  } = Buttons;

  switch (width) {
    case 'lg':
      return {
        button: LargeButton,
        text: LargeText,
      };
    case 'md':
      return {
        button: MediumButton,
        text: MediumText,
      };
    case 'sm':
      return {
        button: SmallButton,
        text: SmallText,
      };
    case 'xs':
      return {
        button: XSmallButton,
        text: XSmallText,
      };
    default:
      return {
        button: LargeButton,
        text: LargeText,
      };
  }
}

const Button = (props) => {
  const { title } = props;
  const Components = getComponents(props);
  return (
    <Components.button {...props}>
      <Components.text>{title}</Components.text>
    </Components.button>
  );
};

export default Button;
