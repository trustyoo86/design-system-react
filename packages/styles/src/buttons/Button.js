/** @jsx jsx */
import { css } from '@emotion/core';

export const button = css`
  width: 100%;
  height: 30px;
`;
