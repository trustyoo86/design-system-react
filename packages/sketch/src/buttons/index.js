'use strict';

import React from 'react';
import { render, Page, Artboard } from 'react-sketchapp';
import Button from './Button';

const Buttons = () => {
  return (
    <Page>
      <Artboard name='테스트'>
        <Button title='테스트' />
      </Artboard>
    </Page>
  );
};

export default (context) => {
  render(<Buttons />, context.document.currentPage());
};
