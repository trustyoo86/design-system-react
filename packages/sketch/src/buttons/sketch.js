'use strict';

import React from 'react';
import { render, Page, Artboard, View, Text } from 'react-sketchapp';
import Button from './Button';

const styles = {
  page: {
    width: 2000,
    height: 1000,
    display: 'flex',
    flexDirection: 'row',
  },
  container: {
    width: 2000,
    height: 640,
    marginRight: 20,
  },
  view: {
    width: 300,
    height: 200,
    backgroundColor: '#c6c6c6',
  },
  text: {
    fontSize: 15,
  },
};

const SketchBoard = () => {
  return (
    <Page style={styles.page}>
      <Artboard name='테스트' style={styles.container}>
        <Button
          key='lg'
          title='Large Button!'
          width='lg'
          type='primary' />
        <Button
          key='md'
          title='Medium Button!'
          width='md'
          type='primary' />
        <Button 
          key='sm'
          title='Small Button!'
          width='sm'
          type='primary' />
        <Button 
          key='xs'
          title='XSmall Button!'
          width='xs'
          type='primary' />
      </Artboard>
    </Page>
  );
};

export default () => {
  render(<SketchBoard />, context.document.currentPage());
};
