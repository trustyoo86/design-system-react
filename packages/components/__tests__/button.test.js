import React from 'react';
import { Button } from '_src/buttons';

const shallow = global.shallow;

describe('[Button] 컴포넌트 테스트', () => {
  it('Button 컴포넌트가 정상적으로 렌더링 되어야 한다.', () => {
    const wrapper = shallow(
      <Button>Test Button!</Button>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
