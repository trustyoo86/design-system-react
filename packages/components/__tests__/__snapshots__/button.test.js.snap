// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Button] 컴포넌트 테스트 Button 컴포넌트가 정상적으로 렌더링 되어야 한다. 1`] = `
<button
  styles={
    Object {
      "map": undefined,
      "name": "1au6vmc",
      "next": undefined,
      "styles": "
  width: 100%;
  height: 30px;
",
      "toString": [Function],
    }
  }
>
  Test Button!
</button>
`;
